user-guide/docs/conclavepass-sample.md

```diff
@@ -1,223 +1,200 @@
 
-This approach is obviously very secure but comes with some limitations:
+This approach is secure but has some limitations:
 
-* If the database grows to a large size then it may take significant network
-  bandwidth/storage on the endpoint device.
+* If the database grows large, it may take significant network bandwidth/storage on the endpoint device.
 * There is no opportunity for sharing or collaboration on database entries.
 
-Now, in reality, for password managers these are not real limitations, but for
-the purposes of this demonstration project we can see that a Conclave Cloud
-based implementation can be used to solve these two issues by accessing the
-database inside an SGX enclave in the cloud.
+For password managers, these are not actual limitations, but this demo project shows that you can use Conclave Cloud 
+to solve these two issues in other types of applications by accessing the database inside an SGX enclave on the cloud.
+
+## What's the architecture of the ConclavePass app?
 
-## How is ConclavePass architected?
+The following diagram illustrates the architecture of the ConclavePass application.
 
-A diagram showing the architecture of the project is shown below. Each component
-in the diagram is named corresponding to the directories in this repository that
-contain the implementation for the component.
+![An architecture diagram of the ConclavePass sample application](assets/ConclavePassArchitecture.drawio.png)
 
-![](assets/ConclavePassArchitecture.drawio.png)
+The components in the diagram are named corresponding to the Conclave Pass directory that contains its implementation.
 
-Each component is described below:
+These components are described below:
 
 ### Functions
-The functions consist of a typescript component that provides methods that can
-be used to add, query, get and remove entries within the user's database. This
-requires access to the unencrypted database and therefore, to ensure privacy,
-runtime encryption is used by hosting the functions within Conclave Cloud in the
-Conclave Functions service.
-
-Conclave Functions are stateless and do not have access to any persistent
-storage. Therefore, external storage within the cloud is necessary in order to
-persist the user databases. The functions component uses the JavaScript
-`fetch()` built-in capability to query and update an external data store with
-each user's encrypted database entries. It is vitally important to ensure that
-any data exchanged via `fetch()` is encrypted as the request is made outside the
-Conclave Functions Intel SGX enclave. The functions module uses another built-in
-function, `crypto.getProjectKey()` to get a key that is unique to the project and
-function code and uses this to encrypt/decrypt the user's database prior to
-exchanging it with the external storage.
+
+The functions component consists of a typescript that provides methods that you can use to add, query, get, 
```

Comment:
"a typescript that..." doesn't read correctly. Maybe "a typescript module that..."